### ISSUE TITLE
[Feature Request] Possibility to use an rsync target

### ISSUE THREAD

**Issue by Germar** [bit-team/backintime#224]
Please implement a function to use remote targets without the need of prior mount by the using the rsync protocol.
- Many NAS devices for home use already have an rsync deamon implemented
- rsync saves much network bandwidth when doing differential backups

---

Imported from Launchpad using lp2gh.
- date created: 2010-08-18T18:18:41Z
- owner: mfm
- the launchpad url was https://bugs.launchpad.net/bugs/620023

**Comment by Germar** [comment]
(by danleweb)
You are right about rsync, but BIT also need to create some special files so it must mount it.
Also BIT must mount it in order to explore snapshots.

**Comment by Germar** [comment]
(by danleweb)
I'll pass it to wishlist ...

**Comment by Germar** [comment]
(by iromeister)
Isn't it possible to explore snapshots via rsync? And even more efficiently than over a network share? Just my 2 ct

**Comment by Germar** [comment]
(by danleweb)
I found only this: http://sourceforge.net/projects/rsyncfs/

**Comment by emtiu** [comment]
> (by danleweb) You are right about rsync, but BIT also need to create some special files so it must mount it. Also BIT must mount it in order to explore snapshots.

Closing as not practical because of this limitation.

**Comment by emtiu** [comment]
This might be doable, after all. Some functions could easily be replaced by rsync-only commands:
- prepare the special files before or after transfer, then include them in another rsync call
- list existing snapshots through some sort of `rsync --dry-run` call?

Other things would be much more tricky, such as:
- deleting or renaming specific snapshots (including smart remove)

Let's keep this open as a reminder. You sometimes get rsync-only targets without ssh access at hosting services, or with branded NAS hardware. As things stand, backintime can't be used with those.

**Comment by buhtz** [comment]
Isn't this related or the same as #213 ?

**Comment by emtiu** [comment]
No, this is about using a remote backup destination where yo **do** have rsync connectivity, but **no** SSH access.

I'm not sure it's realistic with the way that backintime is designed, and demand seems to be low.

**Comment by buhtz** [comment]
> No, this is about using a remote backup destination where yo **do** have rsync connectivity, but **no** SSH access.
> 
> I'm not sure it's realistic with the way that backintime is designed, and demand seems to be low.

Ah I see. In that case it isn't such a big deal, but also not a tiny one. I vote to keep it open.

**Comment by buhtz** [comment]
> I vote to keep it open.

😆 Now, after more than one year, I vote to close it. I don't see this will be implemented someday, nor it is possible to implement in the current codebase because of its structure and code quality.

**Comment by buhtz** [comment]
Closing this ticket based on the comment above. Feel free to reopen
if the problem still exists. Thank you for your efforts. If you have
any further questions, ideas or encounter any other issues, please
don't hesitate to let us know.

Best regards,
